### ISSUE TITLE
Rule proposal: validate PropTypes aren't typos

### ISSUE THREAD

**Issue by jimbolla** [yannickcr/eslint-plugin-react#213]
Rule would check that calls to React.PropTypes.whatever are valid. For instance, I have a bad habit of typing `disabled: PropTypes.bools.isRequired` instead of `disabled: PropTypes.bool.isRequired`. I've also done `disabled: propTypes.bool.isRequired` and `disabled: propTypes.bool.isRequired()` by accident.

**Comment by mik01aj** [comment]
+1, I have the same problem. I also type `React.propTypes` instead of `React.PropTypes`,  `function` instead of func`, `boolean` instead of `bool` and I'm sometimes tempted to type `obj` instead of `object`.

**Comment by willklein** [comment]
:+1: I have done this a lot. The abbreviations are inconsistent.

I was presenting on how to write lint rules at my React user group last night, and we live coded a first pass: https://github.com/willklein/scaling-best-practices/commit/901534ba2cdb2f4a1de190e018d4a5bb4b1c9833

Needs more test cases and a few conditions. I'll see if I can get this wrapped up in the next week.

**Comment by lencioni** [comment]
@willklein have you made any progress on wrapping up this rule?

**Comment by samsch** [comment]
Bumping an old issue with a new case:
```jsx
class A extends React.Component {
  render() {
    return <a href={this.props.href} >{this.props.children}</a>;
  }
}
A.PropTypes = {
  href: React.PropTypes.string.isRequired,
  children: React.PropTypes.node.isRequired,
}
```
This code will be linted by `react/prop-types` as missing prop types for href and children. The subtle problem here is that `A.PropTypes = ...` should be `A.propTypes = ...`.

Could this rule find and warn/error on a case where prop types are being set in a component property other than propTypes?
